### PR TITLE
Update endpoints with new user permission checks

### PR DESCRIPTION
**Pull request by juliasaveliff**
Done in this PR
- [x] Model endpoints
- [x] Asset endpoints (for global permission only)
- [x] Audit log endpoints
- [x] User endpoints

To do in other PR
- [ ] Power endpoints (all validation is currently done on front end)
- [ ] Asset endpoints for datacenter-specific permission

### CODE REVIEW

File: rackcity/permissions/permissions.py [hunk 1]
```diff
@@ -21,6 +21,14 @@ class PermissionCodename(Enum):
     USER_WRITE = 'user_write'
 
 
+class PermissionPath(Enum):
+    MODEL_WRITE = 'rackcity.' + PermissionCodename.MODEL_WRITE.value
+    ASSET_WRITE = 'rackcity.' + PermissionCodename.ASSET_WRITE.value
+    POWER_WRITE = 'rackcity.' + PermissionCodename.POWER_WRITE.value
+    AUDIT_READ = 'rackcity.' + PermissionCodename.AUDIT_READ.value
+    USER_WRITE = 'auth.' + PermissionCodename.USER_WRITE.value
```
Comment (juliasaveliff) [inline rackcity/permissions/permissions.py]: using these for permission_required decorators

File: rackcity/views/asset_views.py [hunk 2]
```diff
@@ -60,20 +62,6 @@
...
-    """
-    List all assets.
-    """
-    if request.method == 'GET':
-        assets = Asset.objects.all()
-        serializer = RecursiveAssetSerializer(assets, many=True)
-        return JsonResponse({"assets": serializer.data})
-
-
-# @close_old_connections_decorator
```
Comment (benauriemma) [inline rackcity/views/asset_views.py]: thx for cleaning up

File: rackcity/views/asset_views.py [hunk 3]
```diff
@@ -168,7 +156,6 @@ def asset_many(request):
     )
 
 
-# @close_old_connections_decorator
```
Comment (benauriemma) [inline rackcity/views/asset_views.py]: ditto

File: rackcity/views/asset_views.py [hunk 4]
```diff
@@ -60,20 +62,6 @@
 from rackcity.models.asset import get_next_available_asset_number
 
 
-# @close_old_connections_decorator
-@api_view(['GET'])  # DEPRECATED !
```
Comment (juliasaveliff) [inline rackcity/views/asset_views.py]: I deleted all deprecated function views & comments to clean up some of these view files

File: rackcity/views/user_views.py [hunk 5]
```diff
@@ -89,7 +91,7 @@ def netid_login(request):
 
 
 @api_view(['POST'])
-@permission_classes([IsAdminUser])
+@permission_required(PermissionPath.USER_WRITE.value, raise_exception=True)
```
Comment (benauriemma) [inline rackcity/views/user_views.py]: Do you need USER_WRITE permission to view users?

File: rackcity/views/asset_views.py [hunk 6]
```diff
@@ -1047,7 +1028,7 @@ def asset_bulk_export(request):
 
 
 @api_view(['POST'])
-@permission_classes([IsAdminUser])
+@permission_required(PermissionPath.ASSET_WRITE.value, raise_exception=True)
```
Comment (juliasaveliff) [inline rackcity/views/asset_views.py]: might need to check on Piazza whether network bulk actions count as asset actions, but I would think that they do

File: rackcity/views/user_views.py (hunk 5 above)
Comment (juliasaveliff) [reply rackcity/views/user_views.py]: oh good catch. technically not write permission, but only admin users can view other users, and only admin users have user write permission, so I think I will keep this;

Comment (benauriemma) [reply rackcity/views/user_views.py]: Okay. @juliasaveliff

**Pull request merged**